packages/flutter/lib/src/gestures/events.dart

```diff
@@ -1986,6 +1986,341 @@ class _TransformedPointerScrollEvent extends _TransformedPointerEvent with _Copy
+    : assert(original != null), assert(transform != null);
+
+  @override
+  final PointerPanZoomStartEvent original;
+
+  @override
+  final Matrix4 transform;
+
+  @override
+  PointerPanZoomStartEvent transformed(Matrix4? transform) => original.transformed(transform);
+}
+
+mixin _CopyPointerPanZoomUpdateEvent on PointerEvent {
+  /// The total pan offset of the pan/zoom
+  Offset get pan;
+  /// The total pan offset of the pan/zoom, transformed into local coordinates
+  Offset get localPan;
+  /// The amount the pan offset changed since the last event
+  Offset get panDelta;
+  /// The amount the pan offset changed since the last event, transformed into local coordinates
+  Offset get localPanDelta;
+  /// The scale (zoom factor) of the pan/zoom
+  double get scale;
+  /// The amount the pan/zoom has rotated in radians so far
+  double get rotation;
+
+  @override
+  PointerPanZoomUpdateEvent copyWith({
+    Duration? timeStamp,
+    int? pointer,
+    PointerDeviceKind? kind,
+    int? device,
+    Offset? position,
+    Offset? delta,
+    int? buttons,
+    bool? obscured,
+    double? pressure,
+    double? pressureMin,
+    double? pressureMax,
+    double? distance,
+    double? distanceMax,
+    double? size,
+    double? radiusMajor,
+    double? radiusMinor,
+    double? radiusMin,
+    double? radiusMax,
+    double? orientation,
+    double? tilt,
+    bool? synthesized,
+    int? embedderId,
+    Offset? pan,
+    Offset? localPan,
+    Offset? panDelta,
+    Offset? localPanDelta,
+    double? scale,
+    double? rotation,
+  }) {
+    return PointerPanZoomUpdateEvent(
+      timeStamp: timeStamp ?? this.timeStamp,
+      kind: kind ?? this.kind,
+      device: device ?? this.device,
+      position: position ?? this.position,
+      embedderId: embedderId ?? this.embedderId,
+      pan: pan ?? this.pan,
+      panDelta: panDelta ?? this.panDelta,
+      scale: scale ?? this.scale,
+      rotation: rotation ?? this.rotation,
+    ).transformed(transform);
+  }
+}
+
+/// The active pan/zoom on this pointer has updated.
+///
+/// See also:
+///
+///  * [Listener.onPointerPanZoomUpdate], which allows callers to be notified of these
+///    events in a widget tree.
+class PointerPanZoomUpdateEvent extends PointerEvent with _PointerEventDescription, _CopyPointerPanZoomUpdateEvent {
+  /// Creates a pointer pan/zoom update event.
+  ///
+  /// All of the arguments must be non-null.
+  const PointerPanZoomUpdateEvent({
+    Duration timeStamp = Duration.zero,
+    PointerDeviceKind kind = PointerDeviceKind.mouse,
+    int device = 0,
+    int pointer = 0,
+    Offset position = Offset.zero,
+    int embedderId = 0,
+    this.pan = Offset.zero,
+    this.panDelta = Offset.zero,
+    this.scale = 1.0,
+    this.rotation = 0.0,
+    bool synthesized = false,
+  }) : assert(timeStamp != null),
+       assert(kind != null),
+       assert(device != null),
+       assert(position != null),
```

Comment:
Why not assert that all of the parameters are not null?  Here and elsewhere.

Reply:
Can you remind me why we assert these non-nullable variables as non-null again?

Reply:
Ah this came up again recently.  So we do need the asserts for now, and the reason is for consumers of this file that are not yet using null safety.